pkg/scale/decode_test.go

```diff
@@ -11,6 +11,7 @@ import (
 
 	"github.com/google/go-cmp/cmp"
 	"github.com/google/go-cmp/cmp/cmpopts"
+	"github.com/stretchr/testify/assert"
```

Comment:
Same question as in #2741 @timwu20 can we add those testing deps? Or should we not? Or should we remove them later? 🤔

Reply:
Since you wrote the book on go testing here at ChainSafe, I don't mind using testify packages in test code for ChainSafe repos.

Reply:
Actually the testing section is still drafty/wip 😄 So nothing official yet!
To be honest, I like having libraries without any dependency, it adds security/ease of mind for library users.
But then yeah writing tests becomes more of pain. I would suggest we can pay attention to it when migrating scale outside Gossamer later on, it shouldn't be hard to remove the dep on testify packages. (also Go needs dev/test-only dependencies imo)